### PR TITLE
Update governance with new roles

### PR DESCRIPTION
**Pull request by jacobweinstock**
## Description

Renamed committers to approvers to follow the pattern established
in Kubernetes community. Added the reviewers role to create a
better latter toward making changes to repositories. Added the
Admin role to help with GitHub administration duties.
This clarifies what inactivity is and how it's measured.

rendered version: https://github.com/jacobweinstock/proposals/blob/0024-community-membership/proposals/0024/GOVERNANCE.md

## Why is this needed

Fixes: #

## Checklist:

I have:

- [ ] updated the documentation and/or roadmap (if required)
- [ ] added unit or e2e tests
- [ ] provided instructions on how to upgrade

-----
[View rendered proposals/0024/GOVERNANCE.md](https://github.com/jacobweinstock/proposals/blob/0024-community-membership/proposals/0024/GOVERNANCE.md)
[View rendered proposals/0024/README.md](https://github.com/jacobweinstock/proposals/blob/0024-community-membership/proposals/0024/README.md)

### CODE REVIEW

File: proposals/0024/GOVERNANCE.md [hunk 1]
```diff
@@ -102,8 +102,41 @@ The following apply to the repo or the part of the codebase for which one would
...
+Therefore members with an extended period away from the project with no activity
+will be removed from the Tinkerbell Github Organization and will be required to
+go through the org membership process again after re-familiarizing themselves
+with the current state.
+
+### How inactivity is measured
+
+Inactive members are defined as members of the Tinkerbell Organization
+with **no** contributions across any repos within 8 months. This is
+measured by the CNCF [DevStats project].
```
Comment (micahhausler) [inline proposals/0024/GOVERNANCE.md:223]: Do we have any mechanism for this? Do we have a job we can run periodically to diff a list of usernames vs recent contributions? Maybe we don't auto-remove, but even if its a monthly job that just scans for activity and reports inactive contributors that might help make this more actionable. 

Maybe we need a list of folks in each role in a yaml file somewhere (Another case to resurrect `github.com/tinkerbell/org`?)

Comment (micahhausler) [reply proposals/0024/GOVERNANCE.md:223]: Opened tinkerbell/org#7

File: proposals/0024/GOVERNANCE.md [hunk 2]
```diff
@@ -102,8 +102,41 @@ The following apply to the repo or the part of the codebase for which one would
...
+
+Therefore members with an extended period away from the project with no activity
+will be removed from the Tinkerbell Github Organization and will be required to
+go through the org membership process again after re-familiarizing themselves
+with the current state.
+
+### How inactivity is measured
+
+Inactive members are defined as members of the Tinkerbell Organization
+with **no** contributions across any repos within 8 months. This is
```
Comment (micahhausler) [inline proposals/0024/GOVERNANCE.md:222]: We can stick with 8 months, but it won't be easy without tooling to easily tell this. The closest time ranges the [CNCF DevStats Activity page for Tinkerbell](https://tinkerbell.devstats.cncf.io/d/66/developer-activity-counts-by-companies?orgId=1&var-period_name=Last%20year&var-metric=contributions&var-repogroup_name=All&var-country_name=All&var-companies=All) has are either the last year or last quarter.
Screen Shot 2021-10-06 at 9 02 54 AM

File: proposals/0024/GOVERNANCE.md [hunk 3]
```diff
@@ -102,8 +102,41 @@ The following apply to the repo or the part of the codebase for which one would
...
+measured by the CNCF [DevStats project].
+
+**Note:** Devstats does not take into account non-code contributions. If a
```
Comment (micahhausler) [inline proposals/0024/GOVERNANCE.md:225]: [Devstats](https://tinkerbell.devstats.cncf.io/d/66/developer-activity-counts-by-companies?orgId=1&var-period_name=Last%20year&var-metric=contributions&var-repogroup_name=All&var-country_name=All&var-companies=All) does have the following metrics:

- Comments
- Commit comments
- Commits
- GitHub Events
- GitHub pushes
- Issue comments
- Issues
- PRs
- Merged PRs
- Review comments
- Contributions

Should we be more clear that any of the above are included? What about mailing list or community meeting activity? Under this model, is that good enough?

**Comment by tstromberg** [comment]
@jacobweinstock - what do you think about switching to a time duration that is easily observable? 

https://github.com/google/pullsheet can do arbitrary values, but it requires someone to run it locally.

**Comment by jacobweinstock** [comment]
Hey @tstromberg, I've played around with https://github.com/google/pullsheet. I think I can make the time duration different than what cncf dev stats gives us. Let me play with pullsheet a bit to see what numbers I can get.

File: proposals/0024/GOVERNANCE.md (hunk 3 above)
Comment (jacobweinstock) [reply proposals/0024/GOVERNANCE.md:225]: yes, more clear is a great idea. Let me rethink this a bit.

**Comment by jacobweinstock** [comment]
update:

I've found some existing scripts that will query devstats with arbitrary time ranges.
see these:
- https://www.giters.com/cncf/devstats/issues/309
- https://github.com/cncf/devstatscode/blob/master/devel/api_dev_act_cnt_repos.sh

I've modified it a touch to work with tinkerbell repos. This will allow us to query for 8 months.
see here:
https://gist.github.com/jacobweinstock/f63f2dc9f32fe00a0aab08bcd7798bf5

**Comment by jacobweinstock** [comment]
I've created a doc outlining one possible way we can go about bootstrapping the roles described in this proposal. I will also bring this up at the next community meeting.

https://docs.google.com/document/d/1Y4RPFZ0smaSLVmK178Q7fStrLQhdi8Y1BrmRmvZHWkU/edit?usp=sharing

File: proposals/0024/GOVERNANCE.md [hunk 4]
```diff
@@ -102,8 +103,135 @@ The following apply to the repo or the part of the codebase for which one would
...
+
+A core principle in maintaining a healthy community is encouraging active
+participation. It is inevitable that people's focuses will change over time and
+they are not expected to be actively contributing forever.
+
+However, being a member of the Tinkerbell GitHub organization comes with
+an [elevated set of permissions]. These capabilities should not be used by those
+that are not familiar with the current state of the Tinkerbell project.
+
+Therefore members with an extended period away from the project with no activity
```
Comment (micahhausler) [inline proposals/0024/GOVERNANCE.md:214]: Could be worth adding a boomerang clause. Life happens and people might have to step away, but given certain timeframes, we could always add them back

**Comment by micahhausler** [comment]
This has been open for 6 months. Given that there have been no objections, I propose that any other required reviewers approve as-is, and we iterate on boomerang clause/role bootstraping in a successive PR/revision

**Comment by displague** [comment]
We'll have to revise https://github.com/orgs/tinkerbell/teams to best fit this structure post-merge.  
Anything else that you are aware of @jacobweinstock, @nshalman?

File: proposals/0024/GOVERNANCE.md (hunk 4 above)
Comment (displague) [reply proposals/0024/GOVERNANCE.md:214]: I think the statement provided in the PR works as a boomerang if we change the two occurrences of `will` to `may`.

File: proposals/0024/GOVERNANCE.md (hunk 2 above)
Comment (displague) [reply proposals/0024/GOVERNANCE.md:222]: I agree that 8 months is somewhat arbitrary and hard to administer value. I'd be fine with a year but I'm also fine to let this stand as worded.

Comment (displague) [reply proposals/0024/GOVERNANCE.md:222]: > [](https://github.com/tinkerbell/proposals/pull/50/files#diff-7f2439d5f7cdc18509ed1616be57259b19699dfd46b3b1675c469ee2289a8ce2)[](https://github.com/)measured by the CNCF [DevStats project].

Perhaps 1 year would be best. Again, this is not a blocker.

File: proposals/0024/GOVERNANCE.md (hunk 3 above)
Comment (displague) [reply proposals/0024/GOVERNANCE.md:225]: What I see is that, before this non-code clause, we are asking for "contributions" without calling out what they are specifically. We can always tighten that up later. I think the current wording is fine for now.

**Review by displague — APPROVED** [review]

**Review by displague — APPROVED** [review]
The age of this PR and the hands that it has passed through indicate that this PR broadly carries the form of clarification we need on the contributor model to foster a healthy ecosystem. 

This PR is an incremental improvement and formalization of the proposed model that we have been guided by for the past 8 months.

We will need to onboard processes to truly enact this proposal. I expect that we may need to revise the proposal as we do so.


As a CNCF project, there are some patterns that we could follow found in adjacent projects and templates, such as https://github.com/cncf/project-template (contributor ladder and governance-maintainer). While we may draw further inspiration from these projects, what we have here is well modeled after the philosophies that make these other open-source projects successful.

**Pull request merged**